File: NetBeansProjects/WebApplication1/src/java/com/epic/canvascontrollibrary/CCLVirtualKeyboardProps.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package com.epic.canvascontrollibrary;

import java.util.ArrayList;
import java.util.List;

/**
 *
 * @author Gilgamesh
 */
    public class CCLVirtualKeyboardProps
    {
        public String CanvasID;
        public String WindowID;
        public String X;
        public String Y;
        public String Width;
        public String Height;
        public List<Object> Keys;
        public String KeyPressFunction;
        public String GapBetweenButtons;
        public String GapBetweenRows;
        public String CurrentKeyboardIndex;
        public List<Object> KeyExtents;
        public String TextHeight;
        public String TextFontString;
        public String CustomKeys;
        public String CustomDrawLetterFunction;
        public String HasGloss;
        public String ShiftKeyPressed;
        public String TabStopIndex;

        CCLVirtualKeyboardProps()
        {
            Keys = new ArrayList<Object>();
            KeyExtents = new ArrayList<Object>();
        }
    }

File: ASP.NET MVC Web Api New Solution/CanvasControlLibrary/CanvasControlLibrary/Models/CCLVirtualKeyboardProps.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.ComponentModel.DataAnnotations.Schema;

namespace CanvasControlLibrary.Models
{
    public class CCLVirtualKeyboardProps
    {
        public class Keyboard
        {
            public class Key
            {
                public string KeyLabel { get; set; }
                public string Width { get; set; }
                public string Height { get; set; }
                public int ReferencingKeyboardIndex { get; set; }
            }
            public List<Key> Keys { get; set; }
        }

        public class KeyExtent
        {
            public int X { get; set; }
            public int Y { get; set; }
            public int Width { get; set; }
            public int Height { get; set; }
            public string KeyLabel { get; set; }
        }

        public string CanvasID { get; set; }
        public int WindowID { get; set; }
        public int X { get; set; }
        public int Y { get; set; }
        public int Width { get; set; }
        public int Height { get; set; }
        public List<Keyboard> Keys { get; set; }
        public int GapBetweenButtons { get; set; }
        public int GapBetweenRows { get; set; }
        public int CurrentKeyboardIndex { get; set; }
        public List<KeyExtent> KeyExtents { get; set; }
        public int TextHeight { get; set; }
        public string TextFontString { get; set; }
        public string CustomKeys { get; set; }
        public int HasGloss { get; set; }
        public int ShiftKeyPressed { get; set; }
        public object Tag { get; set; }
        public int TabStopIndex { get; set; }

        public CCLVirtualKeyboardProps()
        {
            Keys = new List<Keyboard>();
            KeyExtents = new List<KeyExtent>();
        }
    }
}